api/services/search.service.js

```diff
@@ -0,0 +1,30 @@
+import { Conflict } from "../errors/Conflict.js";
+import * as searchRepository from "../repositories/search.repository.js";
+import { UnexpectedError } from "../errors/UnexpectedError.js";
+export const getProductsAndStores = async (req, res, next) => {
+    var { query } = req.body;
+    if(query < 3){
+        return next(Conflict.invalidQuery());
+    }
+    query = query.toString();
+    var searchedData;
+    try{
```

Comment:
whats the try catch for? if its to catch the random error then we could try to create another wrapper to catch these random errors, look at how toe validation wrapper is being used for inspiration